doc(ha): add doc for remount when volume read only (#863)
longhorn/website: content/docs/1.6.0/high-availability/recover-volume-when-read-only.md

---
  title: Recover Volume When It Becomes Read Only
  weight: 1
---

> The state of a volume can change to read-only when IO errors occur. IO errors can be caused by network disconnection, high disk latency, abnormal node shutdown, and other issues.

Reviewer: Suggest briefly describing the cases. 

network disconnection: between engine and replicas
high-disk latency: latency between replica and the corresponding disk
abnormal node shutdown: node down where the engine is running.

Author: Fixed.

> The state of a volume can change to read-only when IO errors occur. IO errors can be caused by network disconnection, high disk latency, abnormal node shutdown, and other issues.

Reviewer 2: ```suggestion
The state of a volume can change to read-only when IO errors occur. IO errors can be caused by a variety of issues, including the following:

- Network disconnection: Interrupted connection between the engine and replicas
- High disk latency: Significant delay in the transfer of data between a replica and the corresponding disk
- Abnormal node shutdown: Failure of a node that is running the engine
```

Reviewer: @derekbit are there any missing or incorrect changes about the causes of IO errors? :point_up:

Reviewer 3: > Abnormal node shutdown: Failure of a node that is running the engine

When an engine is somehow shutdown, application and engine are both gone, so the IO error shouldn't happen.

> abnormal node shutdown

`abnormal node shutdown` in my previous comment might be not necessary.

WDYT? @ChanYiLin @innobead

Author: Yeah agree, and I think here we just want to provide some hints about why it happens
But IO errors may be caused by many factors, I don't think we can and need to  list all of them.

Author: Fixed.

---
  title: Recover Volume When It Becomes Read Only
  weight: 1
---

> A volume is converted to read-only state after IO requests encountered IO errors which can be caused by a network disconnection, high disk latency, abnormal node shutdown and so on.

Reviewer 2: ```suggestion
The state of a volume can change to read-only when IO errors occur. IO errors can be caused by network disconnection, high disk latency, abnormal node shutdown, and other issues.
```

> When an unexpected detachment happens, which can happen during a [Kubernetes upgrade](https://github.com/longhorn/longhorn/issues/703), a [Docker reboot](https://github.com/longhorn/longhorn/issues/686), or a network disconnection, the volume could become read only and leads to IO error.

Reviewer 3: ```suggestion
When an unexpected detachment happens
```

A volume is converted to read-only state after IO requests encountered IO errors rather than detachment.
IO errors can be caused by a network disconnection, high disk latency, abnormal node shutdown and so on.

> Longhorn periodically detects the status of the volume and automatically deletes the workload pod if the pod is managed by a controller (e.g. deployment, statefulset, daemonset, etc...) when the volume becomes read only.
> By deleting the pod, its controller restarts the pod and Kubernetes handles volume reattachment and remount.

Reviewer 2: ```suggestion
Longhorn periodically checks the volume state and automatically deletes the workload pod if the pod is managed by a controller (for example, Deployment, StatefulSet, or DaemonSet) when the state changes to read-only. Once the workload pod is deleted, the controller restarts the pod and Kubernetes handles volume reattachment and remounting.
```

> If you don't want Longhorn to automatically delete the workload pod, you can set it in [the setting `Automatically Delete Workload Pod when The Volume Is Detached Unexpectedly`](../../references/settings#automatically-delete-workload-pod-when-the-volume-is-detached-unexpectedly) in Longhorn UI.

Reviewer 2: ```suggestion
To prevent Longhorn from automatically deleting the workload pod, you can disable the setting on the Longhorn UI. For more information, see [Automatically Delete Workload Pod When the Volume Is Detached Unexpectedly](../../references/settings#automatically-delete-workload-pod-when-the-volume-is-detached-unexpectedly).
```

> For the pods that don't have a controller, Longhorn doesn't delete them because if Longhorn does, no one will restart them.
> To recover read only volumes, you would have to manually delete and recreate the pods that don't have a controller.

Reviewer 2: ```suggestion
Longhorn does not delete pods without a controller because such pods cannot be restarted after deletion. To recover read-only volumes, you must manually delete and restart the pods that are not managed by a controller.
```
